biogeochem/EDPhysiologyMod.F90

```diff
@@ -1096,14 +1145,38 @@ subroutine PlantGrowth( currentSite, currentCohort, bc_in )
     !
     ! Units: kgC/year * (year/days_per_year) = kgC/day -> (day elapsed) -> kgC
     ! -----------------------------------------------------------------------------------
+    previous_bl = currentCohort%bl
 
     currentCohort%bl     = currentCohort%bl - currentCohort%leaf_md*hlm_freq_day
     currentcohort%br     = currentcohort%br - currentCohort%root_md*hlm_freq_day
     currentcohort%bsw    = currentcohort%bsw - currentCohort%bsw_md*hlm_freq_day
     currentCohort%bdead  = currentCohort%bdead - currentCohort%bdead_md*hlm_freq_day
     currentCohort%bstore = currentCohort%bstore - currentCohort%bstore_md*hlm_freq_day
-
     
+    if(use_leaf_age==itrue.and. currentCohort%bl>0.0_r8)then
+    	  !advance the leaf ages	  
+	  dExpLeaves_dt=previous_bl *currentCohort%fracExpLeaves/(EDPftvarcon_inst%leaf_long(ipft)&
+	                 *EDPftvarcon_inst%expleaf_flong(ipft))*hlm_freq_day
+	  dYoungLeaves_dt=previous_bl *currentCohort%fracYoungLeaves/(EDPftvarcon_inst%leaf_long(ipft) &
+	                 *EDPftvarcon_inst%youngleaf_flong(ipft))*hlm_freq_day	
+	  dOldLeaves_dt=previous_bl *currentCohort%fracOldLeaves /(EDPftvarcon_inst%leaf_long(ipft) &
+	                 * EDPftvarcon_inst%oldleaf_flong(ipft))*hlm_freq_day			 			 
+	  dSenLeaves_dt=previous_bl *currentCohort%fracSenLeaves / (EDPftvarcon_inst%leaf_long(ipft) &
+	                 * EDPftvarcon_inst%senleaf_flong(ipft))*hlm_freq_day
+	  if(currentSite%dstatus==1)  then !drought periods
+	       dSenLeaves_dt=previous_bl *currentCohort%fracSenLeaves &
+	       /(EDPftvarcon_inst%senleaf_flong(ipft)*EDPftvarcon_inst%senleaf_long_fdrought(ipft) &
+	       * EDPftvarcon_inst%leaf_long(ipft))*hlm_freq_day	  
+	  endif	
+	  currentCohort%fracExpLeaves = (previous_bl *currentCohort%fracExpLeaves-dExpLeaves_dt)&
+	                /currentCohort%bl
+	  currentCohort%fracYoungLeaves = (previous_bl *currentCohort%fracYoungLeaves+dExpLeaves_dt- &
+	               dYoungLeaves_dt)/currentCohort%bl
+	  currentCohort%fracOldLeaves = (previous_bl *currentCohort%fracOldLeaves+dYoungLeaves_dt- &
+	               dOldLeaves_dt)/currentCohort%bl
+	  currentCohort%fracSenLeaves = (previous_bl *currentCohort%fracSenLeaves+dOldLeaves_dt- &
+	               dSenLeaves_dt)/currentCohort%bl	
+    endif
```

Comment:
should we add a check here to ensure that fracExpLeaves + fracYoungLeaves + fracOldLeaves + fracSenLeaves = 1?

Reply:
added the check with new commit.